added: wireless default unit

jeff-cai opened the pull request:
#4

Preliminary wireless unit.
Future development on displaying ip to be considered.

jeff-cai committed:
added: wireless default unit

jeff-cai committed:
Fix: bin requirement arg in init

fixed to iwconfig (before used wireless_tools which is the package)

qdbp commented:
I would encourage the direct use of /sys/ and /proc/ over requiring binaries, both for performance and compatibility. 

One of the (hitherto unstated) stretch goals for this project is to have no binary blob dependencies (with the exception of units like PY3NVGPU which monitor a proprietary subsystem).

jeff-cai commented:
Further development will migrate towards reading from /sys/ and /proc/. In it's current form I think of it as a placeholder unit that returns the intended outcome (mostly for my own uses) that I thought was stable enough to merge to master despite requiring dependencies.

jeff-cai commented:
The [arch wiki](https://wiki.archlinux.org/index.php/Wireless_network_configuration#Manual_setup) also seems to suggest that these packages are the standard for managing wireless networks, so from a compatibility side I think it would be safe. Though it does mention:

>  wireless_tools - _currently deprecated, but still widely supported._

Perhaps updating to depend on `iw` would be better future-proofing.

qdbp commented on py3s_default_units.py:
```diff
@@ -197,6 +197,54 @@ def get_chunk(self):
+        wireless-tools?
+    '''
+
+    def __init__(self, *args, wlan_id='wlp2s0', **kwargs):
```
Do not guess wlan id. It should be a positional argument.

qdbp commented on py3s_default_units.py:
```diff
@@ -197,6 +197,54 @@ def get_chunk(self):
+    SSID, quality
+
+    Requires:
+        iw (probably?)
```
Ascertain which ones.

jeff-cai committed:
fix: device name to positional arg.

qdbp commented:
Please separate the `get_chunk` method into `read` and `format`, as per the new API.

jeff-cai committed:
update: read/format methods

qdbp commented on src/py3s_default_units.py:
```diff
@@ -300,8 +300,11 @@ def format(self, output):
-    Outputs:
-        SSID, quality
+    Output API:
+        'status': up, down, none
```
please add type prefixes, i.e. 'SSID -> s_SSID', etc

qdbp commented on src/py3s_default_units.py:
```diff
@@ -311,42 +314,51 @@ def __init__(self, wlan_id, *args, **kwargs):
+            return output
 
         # Raw output data
         raw_SSID = findall('ESSID:"(.*?)"', out)[0]
```
does this add a leading space to the returned value of SSID?

qdbp commented on src/py3s_default_units.py:
```diff
@@ -311,42 +314,51 @@ def __init__(self, wlan_id, *args, **kwargs):
         # Quality (overall quality of the link)
         raw_quality = int(raw_quality_num) / int(raw_quality_denom)*100
-        quality = "{:3.0f}%".format(raw_quality)
+        output['quality'] = "{:3.0f}%".format(raw_quality) 
```
should output raw floats, not formatted strings; formatting is `format`'s job!

qdbp reviewed (changes requested):
highlighted some problems

jeff-cai committed:
Fix: type prefixes and formatting

jeff-cai committed:
Add pseudo-color in default units

Temporary solution, fix repetition later

jeff-cai committed:
Add generic color function

jeff-cai committed:
Update color function in wireless module

jeff-cai committed:
Update style to resemble other existing modules

jeff-cai committed:
Minor change to color function, style

qdbp commented on src/py3s_default_units.py:
```diff
@@ -296,6 +298,73 @@ def format(self, output):
+
+    Output API:
+        's_status': up, down, none
+        's_SSID'
```
be more descriptive: SSID of what? the currently connected-to network? one with the strongest signal? etc.

qdbp commented on src/py3s_default_units.py:
```diff
@@ -296,6 +298,73 @@ def format(self, output):
+                wireless device name from `ip link`
+                example: wlan0
+        """
+        self.wlan_id=wlan_id
```
PEP8. (applies in other places too)

qdbp commented on src/py3s_default_units.py:
```diff
@@ -296,6 +298,73 @@ def format(self, output):
+        out = check_output(['iwconfig', self.wlan_id]).decode('ascii')
+        line1 = out.split('\n')[0]
+
+        output = {'s_status': "up", "s_SSID": "SSID", "f_quality": 0}
```
"SSID" is a silly default. I recommend following the convention of the other unit and introducing some `b_` error flags for when this unit's output is not applicable; i.e. there is no connection.

jeff-cai committed:
Apply PEP8

>tfw someone reads these commit messages

jeff-cai committed:
Change color function to bisect

Appears to be buggy though, does not display any output.
Troubleshooting required still
